src/main/java/com/anaplan/client/AnaplanService.java

```diff
@@ -38,15 +38,37 @@
     ACTION
   }
 
-  /** Interface for the CDF plugin that use Anaplan service. */
-  public interface AnaplanConfig {
-    URI getServiceLocation() throws URISyntaxException;
+  /** Wrapper for the metadata for the connection with Anaplan server. */
+  public static class AnaplanConfig {
+    private URI serviceLocation;
+    private URI authServiceLocation;
+    private String username;
+    private String password;
 
-    URI getAuthServiceLocation() throws URISyntaxException;
+    public AnaplanConfig(
+      String username, String password, String serviceLocation, String authServiceLocation)
```

Comment:
Generally good to avoid throwing exceptions in a constructor. Here, it's better to pass in the location variables as URI.

We'll get to it in the plugin PR, but we try to capture multiple validation errors at the same time and associate it with the problematic field. Parsing both here loses some context, as you don't know if the exception is because the service location is invalid or if the auth service location is invalid.

Reply:
Thanks for the another instruction! updated!

Yeah, the concern makes sense. We can talk more in the Plugin PR. But started from here ->
For example, in the action plugin, we will call `config.validate(collector)` in configurePipeline(). Does it makes sense if we also run the validate() in the beginning of the `run()`? I dont see this pattern in the existing plugins, but if we do that, the fields will be validated with the context separately. But, we still does not solve the context losing issue in the scope of the AnaplanConfig. Do we much have it as well?
Thanks! :)